digicampipe/utils/pulse_template.py

```diff
@@ -2,41 +2,123 @@
 import numpy as np
 from scipy.interpolate import interp1d
 
+from digicampipe.io.event_stream import calibration_event_stream
+from digicampipe.calib.baseline import fill_digicam_baseline, \
+    subtract_baseline, correct_wrong_baseline
+from digicampipe.calib.time import estimate_time_from_leading_edge
+from digicampipe.utils.hist2d import Histogram2dChunked
 
-class NormalizedPulseTemplate:
-    def __init__(self, amplitude, time):
-
-        dt = np.diff(time)
-
-        if not (dt == dt[0]).all():
-            raise ValueError('time argument should be of constant sampling')
 
+class NormalizedPulseTemplate:
+    def __init__(self, amplitude, time, amplitude_std=None):
         self.time = time
         self.amplitude = amplitude
+        if amplitude_std is not None:
+            assert len(amplitude_std) == len(amplitude)
+            self.amplitude_std = amplitude_std
+        else:
+            self.amplitude_std = self.amplitude * 0
         self._template = self._interpolate()
+        self._template_std = self._interpolate_std()
 
     def __call__(self, time, amplitude=1, t_0=0, baseline=0):
-
         y = amplitude * self._template(time - t_0) + baseline
+        return np.array(y)
 
+    def std(self, time, amplitude=1, t_0=0, baseline=0):
+        y = amplitude * self._template_std(time - t_0) + baseline
         return np.array(y)
 
     def __getitem__(self, item):
-
         print(self.amplitude[item], item)
-
         return NormalizedPulseTemplate(amplitude=self.amplitude[item],
                                        time=self.time)
 
     @classmethod
     def load(cls, filename):
+        data = np.loadtxt(filename).T
+        assert len(data) in [2, 3]
+        if len(data) == 2:  # no std in file
+            t, x = data
+            return cls(amplitude=x, time=t)
+        elif len(data) == 3:
+            t, x, dx = data
+            return cls(amplitude=x, time=t, amplitude_std=dx)
 
-        t, x = np.loadtxt(filename).T
-
-        return cls(amplitude=x, time=t)
+    @classmethod
+    def create_from_datafiles(
+            cls, input_files, pixels=(0,), time_range_ns=(-10, 40),
+            amplitude_range=(-.1, 0.4), n_bin=100
+    ):
+        events = calibration_event_stream(input_files)
+        events = fill_digicam_baseline(events)
+        if "SST1M_01_201805" in input_files[0]:  # fix data in May
```

Comment:
This should not be performed here! It does only benefit from the use of this function. 
We should solve this problem at the reading of the file in `zfits.py`